pkgs/tools/misc/coreutils/default.nix

```diff
@@ -75,80 +84,89 @@ stdenv.mkDerivation (rec {
-    ++ optionals stdenv.hostPlatform.isCygwin [ texinfo ];  # due to patch
+  nativeBuildInputs = [
+    # autoreconfHook is due to patch, normally only needed for cygwin
+    autoreconfHook
+    perl
+    xz.bin
+  ]
+  ++ lib.optionals stdenv.hostPlatform.isCygwin [
+    # due to patch
+    texinfo
+  ];
+
   configureFlags = [ "--with-packager=https://nixos.org" ]
-    ++ optional (singleBinary != false)
-      ("--enable-single-binary" + optionalString (isString singleBinary) "=${singleBinary}")
-    ++ optional withOpenssl "--with-openssl"
-    ++ optional stdenv.hostPlatform.isSunOS "ac_cv_func_inotify_init=no"
-    ++ optional withPrefix "--program-prefix=g"
-    ++ optional stdenv.isDarwin "--disable-nls" # the shipped configure script doesn't enable nls, but using autoreconfHook does so which breaks the build
-    ++ optionals (stdenv.hostPlatform != stdenv.buildPlatform && stdenv.hostPlatform.libc == "glibc") [
+    ++ lib.optional (singleBinary != false)
+      ("--enable-single-binary" + lib.optionalString (lib.isString singleBinary) "=${singleBinary}")
+    ++ lib.optional withOpenssl "--with-openssl"
+    ++ lib.optional stdenv.hostPlatform.isSunOS "ac_cv_func_inotify_init=no"
+    ++ lib.optional withPrefix "--program-prefix=g"
+    # the shipped configure script doesn't enable nls, but using autoreconfHook
+    # does so which breaks the build
+    ++ lib.optional stdenv.isDarwin "--disable-nls"
+    ++ lib.optionals (isCross && stdenv.hostPlatform.libc == "glibc") [
       # TODO(19b98110126fde7cbb1127af7e3fe1568eacad3d): Needed for fstatfs() I
       # don't know why it is not properly detected cross building with glibc.
       "fu_cv_sys_stat_statfs2_bsize=yes"
     ];
 
-
   buildInputs = [ gmp ]
-    ++ optional aclSupport acl
-    ++ optional attrSupport attr
-    ++ optional withOpenssl openssl
-    ++ optionals selinuxSupport [ libselinux libsepol ]
-       # TODO(@Ericson2314): Investigate whether Darwin could benefit too
-    ++ optional (stdenv.hostPlatform != stdenv.buildPlatform && stdenv.hostPlatform.libc != "glibc") libiconv;
+    ++ lib.optional aclSupport acl
+    ++ lib.optional attrSupport attr
+    ++ lib.optional withOpenssl openssl
+    ++ lib.optionals selinuxSupport [ libselinux libsepol ]
+    # TODO(@Ericson2314): Investigate whether Darwin could benefit too
+    ++ lib.optional (isCross && stdenv.hostPlatform.libc != "glibc") libiconv;
 
   # The tests are known broken on Cygwin
   # (http://article.gmane.org/gmane.comp.gnu.core-utils.bugs/19025),
   # Darwin (http://article.gmane.org/gmane.comp.gnu.core-utils.bugs/19351),
   # and {Open,Free}BSD.
   # With non-standard storeDir: https://github.com/NixOS/nix/issues/512
   # On aarch64+musl, test-init.sh fails due to a segfault in diff.
-  doCheck = stdenv.hostPlatform == stdenv.buildPlatform
+  doCheck = isCross
     && (stdenv.hostPlatform.libc == "glibc" || stdenv.hostPlatform.isMusl)
```

Comment:
@dasJ about this line: is there any difference between `stdenv.hostPlatform.isMusl` and `stdenv.hostPlatform.libc == "musl"`?

Reply:
I honestly don't know